### ISSUE TITLE
RFC 0004: 10% or 110%

### ISSUE THREAD

**Issue by ligthyear** [maidsafe/rfcs#118]
I'm just going through all the RFCs and make some general typo, grammar and formatting updates, when I found this current section in the [RFC 0004 - Farming](https://github.com/maidsafe/rfcs/blob/3273f133a1c11ff4143799c33ead5d57c2809a61/agreed/0004-Farm-attempt/0004-Farm-attempt.md#rate-attempts) that reads:

```
1. PmidNodes -> tested at 100% of FR [i.e. Farming rate * 1]
2. App Developer -> tested at 10% of FR [i.e. Farming rate * 1.1]
3. Publisher -> tested at 10% of FR [i.e. Farming rate * 1.1]
4. Core development -> tested at 5% of FR [i.e. Farming rate * 1.05]
```

Which is clearly confusing, as 1 describes `100%` as equaling `1`, while two and all following describe the percentage (`10%` or `5%`) as an additional (`* 1.1` or `* 1.05`) – which one is it? I assume that it is meant as additional, so may I propose to the change percentages from 2 down to contain a `+` or the word "additional"?

(If accepted I can make that part of the bigger PR I am planning for the general wording and stuff).

---

I am not sure, where to suggest that, so I'll leave it here for now: I think for many RFCs a "definitions" sections on the top clearly defining words would be very helpful, like in this case what does "Farming" mean? There is a spec but it is totally unclear.

**Comment by dirvine** [comment]
Yes definitions section will help the RFCs.

The farming rate is confusing as currently stated. It is probably wrong to show as a percentage at all actually. It is a rate at which an id is mod divided by this rate to produce a "farming attempt. This means the message has given a node a chance to request a data item (safecoin) be created. If there is currently no item present then this farming attempt will be successful. 

I hope this helps. Generally for all RFCs there is an issue created while in process so here is perfectly fine for any suggestions. 

Thanks again for the input, very much appreciated.

**Comment by ligthyear** [comment]
> Yes definitions section will help the RFCs.

Maybe you could clarify what "farming" means in this particular process? Like does 

> - **Farming** (in this RFC) refers to the process of creating and agreeing upon the creation of new safecoin in the network through a mathematical calculation
> - **Attempt**: as the calculation isn't bound to succeed, the attempt describes when the process will be tried by whom
> - **Farming Rate** is the frequency at which this process be attempted as defined by the network

capture it? (Both terms are still a bit confusing to me).

If so, then what is the "Rate Attempt"-section trying to explain? How often each entity might be try to attempt farming (if so, how is this checked/enforced)? Or how they are awarded on a successful attempt? Or is that an order in which the attempts take place? Maybe one clarifying sentence on top of that section would be helpful.

**Comment by dirvine** [comment]
FArming is the process of obtaining payment for providing resources to the network. A safecoin is a token created by a group close to the address of a safecoin. This address is re result of a hash of data related to a request. This data includes the message itself, the group members managing the node and the nodes address that returns the data. This hash is then modulo divided by the `rate`  when === zero then the attempt can be made. The group send the request to the group of the safecoin. That group then create the token if there is none existing already. The token is created with the node that supplied the data as the owner. 

Not sure if  I am helping here (currently in code), shout if not, perfectly Ok

**Comment by ligthyear** [comment]
No, worries, you can also just do it later, not necessary to do it immediately ;) .

It was helpful. Let me repeat (and correct me if I am wrong): So the calculation (including the framerate) only determines _whether an attempt can be made at all_. An attempt then is _trying to agree_ upon that creation (in the group close to the address) of a safecoin (which is awarded for providing resource). That makes all sense. However, what is the "rate attempt" section about in that scenario?

Does it describe that the calculation for the `PmidNode` takes place every time (at 100% frame rate), while the calculation with the `AppDeveloper` (id? hash?) takes place at (10% of frame rate, one out of ten tries?) and so on?

**Comment by dirvine** [comment]
Yes you are correct. The rate attempt refers to the mod division part. 

There was also the notion of this rate being affected by also including the number of lost data and stored data ratio. So this introduced a further rate. This additional rate was applied to the farm attempt rate. It is no longer done in code though and the RFC will require an update to remove this secondary rate. Good news as it is confusing. 

Now the farming rate is basically only the mod division divisor and this rate is altered by the network itself. So the network increases this rate (divisor) when resources increase (slowing down payments or more precisely, attempts to create a token (safecoin)). This rate (divisor) decreases as resources decrease thereby increasing frequency of payments, encouraging more nodes to join the lottery. 

Thanks again
